ci-operator/step-registry/gather/azure-cli/gather-azure-cli-commands.sh

```diff
@@ -74,7 +73,9 @@ az version
 cat "${TMPDIR}/azure-instance-names.txt" | sort | grep . | uniq | while read -r VM_NAME
 do
   echo "Gathering console logs for ${VM_NAME} in resource group ${RESOURCE_GROUP}"
-  LC_ALL=en_US.UTF-8 az vm boot-diagnostics get-boot-log --name "${VM_NAME}" --resource-group "${RESOURCE_GROUP}" --subscription "${SUBSCRIPTION_ID}" > "${ARTIFACT_DIR}/${VM_NAME}-boot.log"
+  # || true at the end of this line ensures that if boot diagnostics aren't enabled, we don't exit the script.
+  # This allows us to continue and try to gather other boot logs.
+  LC_ALL=en_US.UTF-8 az vm boot-diagnostics get-boot-log --name "${VM_NAME}" --resource-group "${RESOURCE_GROUP}" --subscription "${SUBSCRIPTION_ID}" > "${ARTIFACT_DIR}/${VM_NAME}-boot.log" || true
```

Comment:
you probably want to record the failure, continue to attempt the others, and then exit non-zero if there were any failures.  Otherwise we might not notice "all the boot-log gathering is broken" until someone goes to look and finds it missing.  Like [this][1], so for you, something like:

```sh
EXIT_CODE=0
for VM_NAME in $(sort "${TMPDIR}/azure-instance-names.txt" | grep . | uniq)
do
  echo "Gathering console logs for ${VM_NAME} in resource group ${RESOURCE_GROUP}"
  LC_ALL=en_US.UTF-8 az vm boot-diagnostics get-boot-log --name "${VM_NAME}" --resource-group "${RESOURCE_GROUP}" --subscription "${SUBSCRIPTION_ID}" > "${ARTIFACT_DIR}/${VM_NAME}-boot.log" || EXIT_CODE="${?}"
done

exit "${EXIT_CODE}"
```

[1]: https://github.com/openshift/release/blob/5eae3eb1f102bd25e204648531f7ccb42ee029b2/ci-operator/step-registry/openshift/e2e/test/openshift-e2e-test-commands.sh#L203-L212